src/util/psbt/macros.rs

```diff
@@ -130,6 +130,30 @@ macro_rules! impl_psbt_insert_pair {
     };
 }
 
+#[cfg_attr(rustfmt, rustfmt_skip)]
+macro_rules! impl_psbt_insert_hash_pair {
```

Comment:
Why exactly is this a macro? Couldn't this be a generic function? If possible (without causing redundancy) I think we should avoid macros.

Reply:
I agree. I think this can be turned into a generic function without any redundancy. Will do so